DoSomething/rogue
# History log of events

#### What's this PR do?
Adds a log of up to 20 of the most recent events for a signup. 

- Adds to `EventTransformer` to normalize data. Right now, some `content` objects in the `events` table are returned with `\` (more recent events) while others are not. Because of this, the content column wasn't always being returned an an object but as a string. We want it to always be returned in the same format. 
- When the `HistoryModal` is opened, an api call to the `/events` endpoint is made to get the events data to display in the log. 
- Creates a `parseEventData` function in the `HistoryModal` component to parse through events and make sure we don't return those that nothing actually got updated on the signup (`quantity` or `why_participated`. This is because of the way our events are set up. Right now, if a new post is added, a signup event is created since the post touches the signup but nothing really has changed on the signup properties. 
- Updates `Table` component to be compatible with both events data and campaign data to be used to display both tables (event log and campaign table). 
- Adds  `CampaignRow` and `EventRow` to create different types of rows for the table. 
- Adds `history-modal.scss` so that you can scroll through the modal if there are many events. 
- Screenshot of what the history log looks like: 
![screen shot 2017-10-06 at 10 38 54 am](https://user-images.githubusercontent.com/9019452/31283074-99118fa0-aa82-11e7-8e18-fd6f976a3281.png)

#### How should this be reviewed?
- Make sure you can see the events log on the signup page (`/signups/:sid`), the campaign single page (`/campaigns/:nid`), and the campaign inbox page ('/campaigns/:nid/inbox`). 
- Make sure the events are parsed correctly. 
- For the signup page: 
  - Signup `index.js`'s `showHistory` function make API call to `/events` and passes the signup events to `HistoryModal`.
  - `HistoryModal` parses the events in the `parseEventData` function and sends to the `Table`, along with what `Type` of table to create. 
  - The `Table` sends the data to the `Row` to create the specific rows. 
- For the campaign single and inbox pages: 
  - `WithReviewing` component makes API call `/events` in the `showHistory` function and passes to `CampaignSingle` and `CampaignInbox` pages. Those both pass to `HistoryModal` through a prop. 
  - The rest is the same as above. 

#### Relevant tickets
Fixes [this Pivotal Card](https://www.pivotaltracker.com/n/projects/2019429/stories/150986550)

#### Checklist
